File: Estudo-API/Estudo-API-Casa-Do-Codigo/src/main/java/io/github/monthalcantara/casadocodigo/domain/Autor.java
package io.github.monthalcantara.casadocodigo.domain;

import io.github.monthalcantara.casadocodigo.dto.response.AutorResponseDto;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Builder
@Getter
@Setter
@Entity
@Table(name = "autores")
public class Autor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String email;

    private String nome;

    private String descricao;

    private LocalDate dataCriacao;

    public AutorResponseDto toResponse() {
        return AutorResponseDto.builder()
                .dataCriacao(this.dataCriacao)
                .email(this.email)
                .nome(this.nome)
                .descricao(this.descricao)
                .build();
    }
}

File: Kotlin/API-Casa_Do_Codigo_Kotlin/src/main/kotlin/io/github/monthalcantara/casadocodigo/model/Autor.kt
package io.github.monthalcantara.casadocodigo.model

import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name ="autores")
data class Autor(
    @field:Id @field:GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long? = null,
    @Column(nullable = false) val email: String,
    @Column(nullable = false) val nome: String,
    @Column(nullable = false) val descricao: String,
    val instanteCriacao: LocalDateTime
)

